custom-schemes.md

```diff
@@ -0,0 +1,41 @@
+
+1. Can wallets reliably determine their invoker?
+
+   - Android and iOS offer no facility to determine the web origin which triggered a navigation to a custom URL scheme. 
+   - This seems to increase the opportunity for attacks such as replay attacks, man-in-the-middle attacks and phishing. It also seems to limit the opportunity for abuse detection and prevention.
+   - A browser API can securely pass the requesting origin on to the wallet so that the wallet can know with confidence what origin the user was on when the request was initiated.
+
+2. Can wallets limit requests to secure contexts?
+
+   - If a user uses a website via http (not https) and it invokes a custom scheme, it's possible that an attacker with access to the network could eavesdrop on or even modify the request (such as by replacing the reader public key with their own). 
+   - A browser API could take the position that it must function only within the context of a secure context (https).
+
+3. What is the user experience when multiple wallet apps are installed?
+
+   - When multiple Android applications register for the same custom URL scheme the operating system presents the user with a picker every time it is invoked. Even if the verifier provided the user with appropriate context, a user may have trouble remembering which application has which credentials and so be more likely to cancel or fail the request.
+   - On iOS when multiple applications are installed that register for the same custom URL scheme, the behavior is [undefined](https://stackoverflow.com/questions/13130442/multiple-apps-with-the-same-url-scheme-ios). One wallet will get invoked without any indication that other wallets were available.
+
+4. What is the user experience when no wallet app is installed or when the user cancels wallet invocation?
+
+   - Browsers generally silently ignore navigations to URL schemes with no handlers installed. This likely means a website cannot distinguish between the time it takes for a wallet to process a request and nothing happening due to the lack of any wallet app being installed. 
+   - Additionally browsers on Android and iOS often offer the user the choice of whether or not to invoke an application for a custom URL scheme navigation (even if there is only one such app). The user canceling at this stage is also indistinguishable from no valid application being installed.
+   - A browser-based API could show the user some meaningful error message when no relevant wallets are installed.
```

Comment:
Thanks! Since I'm creating one alternatives analysis here, I can also add some links for CHAPI. I do think we have a lot to learn from it. Maybe we should have a (separate, outside this PR) discussion on the design tradeoffs in CHAPI relative to what we've been discussing for IdentityCredential?

Reply:
See comments in https://github.com/WICG/identity-credential/pull/37#discussion_r1375504345 -- CHAPI and the Identity Credentials API are far more aligned in philosophy and direction than most of the other solutions, in a way that they're already roughly 75%-ish converged, AFAICT.

@samuelgoto has done an excellent job deep-diving on CHAPI and I believe understands it at enough depth to provide better input on IC API's direction than any text could provide. That said, it might not be that difficult to identify the gaps between IC API and CHAPI (I expect that they are few in number).

Reply:
Thanks Manu! Yes I've heard a bunch about CHAPI from Sam and I know it's shaped his thinking on design. I'm personally comfortable that we've done enough here to ensure CHAPI is being leveraged, but I was worried that we perhaps hadn't communicated that well enough to folks who had invested a lot of their time and energy into CHAPI 😊. So perhaps just citing CHAPI in the alternatives section of the README is sufficient?

Reply:
Thanks for the consideration, yes, citing CHAPI (and it's broad alignment w/ Identity Credentials API) in the alternatives section would be sufficient.

Not for this PR, but we might also consider having a matrix comparing the various approaches/protocols/drawbacks. I know those can be difficult to create, but they can also be helpful to get people up to speed w/ what certain solutions do/don't do. I believe it would be fairly easy to do a comparison between IC API, CHAPI, and OID4VCI/VP. Just a thought, not asking anyone to do this right now.

Reply:
Yep sounds good. Thanks!